### PR TITLE
Ask user to confirm the name of the pilot to delete.

### PR DESCRIPTION
**Pull request by quyykk**
**Feature:** This PR implements the feature request detailed and discussed in #6430 

## Feature Details

When deleting a pilot instead of simply a ok/cancel dialog the user is asked to enter the name of the pilot to delete. This prevents any accidental deletions by blindly pressing enter. This has precedence: Github does the same thing when you want to delete a repo.

If the name entered doesn't match then an additional dialog pops up telling the same that the name doesn't match and that the pilot isn't deleted.

## UI Screenshots

safe delete

safe delete no match

## Testing Done

Tried to delete a pilot when I'm blindly pressing enter => doesn't work.
Wrong name entered into the dialog => pilot doesn't get deleted
Correct name entered => pilot is deleted.

## Performance Impact

N/A.

### CODE REVIEW

**Comment by Hurleveur** [comment]
I personally think this is a great improvement, given the fact that otherwise you could just be thinking you're deleting a waypoint of a pilot but you're actually deleting the whole thing, which happened to me, thankfully not with my main save

**Review by petervdmeer — APPROVED** [review]
Looks good.

**Comment by tehhowch** [comment]
I'm not sure how much use it would get, but the UX would be greatly improved if the dialog could not be confirmed unless the provided input matched, rather than disappearing and then being validated.

The user would then be shown the "validation failed" message over the input dialog, rather than need to reopen the modal entirely. Probably their input should be cleared on failure.

There's a number of ways to go about it, but I think an additional constructor that takes a "validator" parameter is a simple way to go. (The validator could be just a `(string) -> bool` callable, or might include a message to display on failure, or trigger some change in UI when the callable returns true, etc.)

**Comment by petervdmeer** [comment]
> Probably their input should be cleared on failure.
Maybe better not, the player likely just made a typo while deleting and keeping the input allows correction of just the typo.

**Comment by tehhowch** [comment]
> Maybe better not, the player likely just made a typo while deleting and keeping the input allows correction of just the typo.

In the interest of increasing friction, I would prefer active engagement that requires careful attention. To that end, we should probably require the user types "delete" as well as the pilot name.

**Comment by quyykk** [comment]
Only requiring the name of the save file should be enough. You need to know what to enter and you do that by reading the dialog so you know you're deleting the pilot.

File: source/LoadPanel.cpp [hunk 1]
```diff
@@ -507,7 +509,7 @@ void LoadPanel::LoadCallback()
 
 
 
-void LoadPanel::DeletePilot()
+void LoadPanel::DeletePilot(const string &pilot)
```
[the remark this answers is not in the document]
Comment (quyykk) [reply source/LoadPanel.cpp:512]: Dialogs with text input require a parameter for the callback for the text just inputted. I can always add a constructor overload, but this seems to be a special case (when the input isn't actually needed) so I don't think it makes sense to add a special overload just for this. Thoughts?

[the remark this answers is not in the document]
Comment (tehhowch) [reply source/LoadPanel.cpp:512]: In that case, drop the variable name to convey it is intentionally unused and only there for the signature.

**Pull request merged**